wallet-new/src/Cardano/Wallet/Kernel/Transactions.hs

```diff
@@ -145,9 +146,10 @@ newTransaction :: ActiveWallet
                -> IO (Either NewTransactionError TxAux, Utxo)
 newTransaction ActiveWallet{..} spendingPassword options accountId payees = do
 
-    -- | NOTE(adn) This number was computed out of the work Matt Noonan did
-    -- on the size estimation. See [CBR-318].
-    let maxInputs = 350
+    -- | NOTE(mn) 65536 is the current maximum transaction size, but this can change
+    --   over time. @newTransaction@ should be parameterized over this value, perhaps
+    --   adding it to @CoinSelectionOptions@.
+    let maxInputs = estimateMaxTxInputs dummyAddrAttrSize dummyTxAttrSize 65536
```

Comment:
In theory, we should be able to read this value directly from the database -- have a look at `verifyGState` and follow the rabbit hole, i think they fetches the current "block meta params" directly from the rocksDB.

It should be relatively straightforward to hook in, and we can be sure we won't need to amend this code further -- leaving the magic constant here is tempting but dangerous, as during the next soft fork (if any) to bump the max Tx size, I'm pretty sure we would forget tweaking this number accordingly 😉

Reply:
That seems reasonable, I'll go down this rabbit hole when I get back on Wednesday morning.

Reply:
As discussed on slack, this will involve some refactoring of `ActiveWallet` to remove the max tx size constant. I'll do this work on a separate PR to keep it logically separated.

Reply:
Don't worry about this for now, this is actually part of a larger problem (CBR-250) which I will take a look at.